### PR TITLE
offline.plot/iplot improvements

### PR DESCRIPTION
**Pull request by jonmmease**
### Overview
This PR includes a variety of small improvements to html generation in `plotly.offline.plot` and `plotly.offline.iplot`.

### Changes
 - Add support for including the auto-resize logic in the HTML div output by `offline.plot` when `output_type='div'` and `include_plotlyjs=False`. See #1043 for discussion. cc @mchels @ryancollingwood @spencercal
 - Adds resize support for iplot use in the classic notebook. This PR now superceeds #970.
 - Adds `'cdn'`, `'directory'`, and path string `include_plotlyjs` options in `offline.plot`
    - When `'cdn'`, the resulting html file/div includes a script tag reference to the plotlyjs cdn. This addresses part of #1227
    - When `'directory'`, the resulting html file/div includes a script tag reference to a plotly.min.js bundle in the same directory as the html file. If output_type is `'file'` then this plotly.min.js bundle is created in the output directory if it doesn't already exist. This is an alternative approach towards the goal of #688
    - When a string ending with `'.js'`, the resulting html file/div includes a script tag that references this exact path. This can be used to point to a plotly.js bundle from an alternative CDN.

### CODE REVIEW

**Comment by jonmmease** [comment]
1494538 Adds resize support for `iplot` use in the classic notebook. This PR now superceeds #970.

**Comment by jonmmease** [comment]
0cbcce7 adds `'cdn'` and `'directory'` `include_plotlyjs` options in `offline.plot`

When `'cdn'`, the resulting html file/div includes a script tag reference to the plotlyjs cdn.  This addresses part of #1227

When `'directory'`, the resulting html file/div includes a script tag reference to a plotly.min.js bundle in the same directory as the html file. If `output_type` is `'file'` then this plotly.min.js bundle is created in the output directory if it doesn't already exist. This is an alternative approach towards the goal of #688

**Comment by jonmmease** [comment]
69dc7ba  Adds option to specify `include_plotlyjs` as a path/url to a *.js file.  This makes it possible to point the resulting html file/div to an alternative CDN location.

**Comment by fu** [comment]
very nice! Thanks for picking this one up 👍

File: plotly/offline/offline.py [hunk 1]
```diff
@@ -389,6 +394,17 @@ def iplot(figure_or_data, show_link=True, link_text='Export to plot.ly',
         ipython_display.display(ipython_display.HTML(script))
 
 
+def _build_resize_script(plotdivid):
+    resize_script = (
+        ''
```
Comment (jonmmease) [inline plotly/offline/offline.py]: Remove empty string here

File: plotly/tests/test_core/test_offline/test_offline.py [hunk 2]
```diff
@@ -21,14 +21,29 @@
...
+
+directory_script = '<script src="plotly.min.js"></script>'
+
 
 class PlotlyOfflineBaseTestCase(TestCase):
     def tearDown(self):
         # Some offline tests produce an html file. Make sure we clean up :)
         try:
             os.remove('temp-plot.html')
+            # Some tests that produce temp-plot.html]
```
Comment (jonmmease) [inline plotly/tests/test_core/test_offline/test_offline.py]: remove ']' character

**Pull request merged**